addons/pinniped/config-controller/controllers/controllers.go

```diff
@@ -92,49 +79,50 @@ func (c *PinnipedController) Reconcile(ctx context.Context, req ctrl.Request) (r
 		log.V(1).Info("pinniped-info configmap not found, setting value to nil")
 		pinnipedInfoCM.Data = nil
 	}
-	// if req is empty, CM changed, let's loop through all clusters and create/update/delete secrets
+	// if req is empty, CM changed, let's loop through all secrets and create/update/delete as needed
 	if (req == ctrl.Request{}) {
-		log.V(1).Info("configmap changed, checking all clusters")
-		clusters := &clusterapiv1beta1.ClusterList{}
-		if err := c.client.List(ctx, clusters); err != nil {
-			log.Error(err, "error listing clusters")
+		log.V(1).Info("empty request provided, checking all secrets")
+		secrets := &corev1.SecretList{}
+		selector := metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{Key: constants.PackageNameLabel, Operator: metav1.LabelSelectorOpIn, Values: []string{"pinniped"}},
+			},
+		}
+
+		s, err := metav1.LabelSelectorAsSelector(&selector)
+		if err != nil {
+			log.Error(err, "failed to convert selector to LabelSelector")
 			return reconcile.Result{}, err
 		}
 
-		for i := range clusters.Items {
-			if utils.IsManagementCluster(&clusters.Items[i]) {
-				log.V(1).Info("skipping reconciliation of management cluster")
-				continue
-			}
+		if err := c.client.List(ctx, secrets, client.MatchingLabelsSelector{Selector: s}); err != nil {
+			log.Error(err, "Error getting Pinniped secret for cluster")
+			return reconcile.Result{}, err
+		}
 
-			if err := c.reconcileAddonSecret(ctx, &clusters.Items[i], pinnipedInfoCM); err != nil {
-				log.Error(err, "error reconciling addon secret")
+		for i := range secrets.Items {
+			if err := c.reconcileSecret(ctx, &secrets.Items[i], pinnipedInfoCM); err != nil {
+				log.Error(err, "error reconciling secret")
 				return reconcile.Result{}, err
 			}
 		}
 		return reconcile.Result{}, nil
 	}
 
-	// Get cluster from rec
-	cluster := clusterapiv1beta1.Cluster{}
-	if err := c.client.Get(ctx, req.NamespacedName, &cluster); err != nil {
+	// Get secret from rec
+	secret := corev1.Secret{}
+	if err := c.client.Get(ctx, req.NamespacedName, &secret); err != nil {
 		if k8serror.IsNotFound(err) {
-			if err := c.reconcileClusterDelete(ctx, req.NamespacedName); err != nil {
-				return reconcile.Result{}, fmt.Errorf("failed to reconcile cluster delete: %w", err)
-			}
+			// If secret not found, assume cluster was deleted, secret will be deleted
+			log.V(1).Info("could not find secret, assuming it has been deleted")
 			return reconcile.Result{}, nil
 		}
-		log.Error(err, "error getting cluster")
+		log.Error(err, "error getting secret")
```

Comment:
Shall we be verbose and log the secret name also?

Reply:
the request object (which includes the name/namespace) has been added to the logs above in this line:
```
log := c.Log.WithName("reconcile").WithValues("request object", req)
```

Reply:
ok cool. ill have to run this again and look at it. I'm not super familiar with the parameterized logging yet 👍